plugin/src/main/java/com/playmonumenta/libraryofsouls/bestiary/BestiarySoulInventory.java

```diff
@@ -174,28 +234,51 @@ private static String formatWell(String in) {
 
 	private final SoulEntry mSoul;
 	private final BestiaryArea mParent;
+	/* TODO: Use these to provide previous / next buttons */
+	private final List<BestiaryEntryInterface> mPeers;
+	private final int mPeerIndex;
+	private int mPrevEntry = -1;
+	private int mNextEntry = 4000;
 
-	public BestiarySoulInventory(Player player, SoulEntry soul, BestiaryArea parent, boolean lowerInfoTier) {
+	public BestiarySoulInventory(Player player, SoulEntry soul, BestiaryArea parent, boolean lowerInfoTier, List<BestiaryEntryInterface> peers, int peerIndex) {
 		super(player, 36, LegacyComponentSerializer.legacySection().serialize(soul.getDisplayName()));
 
 		mSoul = soul;
 		mParent = parent;
+		mPeers = peers;
+		mPeerIndex = peerIndex;
+		mNextEntry = mPeers.size();
 
 		NBTTagCompound vars = soul.getNBT();
 		EntityNBT entityNBT = EntityNBT.fromEntityData(soul.getNBT());
+		AttributeContainer attr = ((MobNBT)entityNBT).getAttributes();
+
 
 		double armor = 0;
 		double armorToughness = 0;
-		double health = vars.hasKey("Health") ? 0.0 + Float.valueOf(vars.getFloat("Health")) : 0;
-		double speed = vars.hasKey("MovementSpeed") ? 0.0 + Float.valueOf(vars.getFloat("MovementSpeed")) : 0.0;
-		double damage = entityNBT != null && entityNBT.getVariable("AttackDamage") != null ? 0.0 + Double.valueOf(entityNBT.getVariable("AttackDamage").get()) : 0.0;
+		double health = vars.hasKey("Health") ? 0.0 + Float.valueOf(vars.getFloat("Health")) : 0.0;
+		double speed = vars.hasKey("MovementSpeed") ? 0.0 + Float.valueOf(vars.getFloat("MovementSpeed")) : 0;
+		double damage = attr.getAttribute(AttributeType.ATTACK_DAMAGE) != null ? attr.getAttribute(AttributeType.ATTACK_DAMAGE).getBase() : 0.0;
 		double speedScalar = 0;
 		double speedPercent = 1;
 		double bowDamage = 0;
 		double explodePower = 0;
-		boolean ranged = false;
-		boolean trident = false;
-		boolean explode = false;
+		double horseJumpPower = 0;
+		double handDamage = 0;
+		DamageType type = null;
+
+		//Stuff to throw errors before everything
+		if (mDefaultHealth.get(entityNBT.getEntityType()) != null && health == 0.0) {
+			health += mDefaultHealth.get(entityNBT.getEntityType());
```

Comment:
This pattern in general should be avoided - use .get() once, and store the value, then use it in both places.